lib/pages/portal/view/website_view.dart

```diff
@@ -134,15 +134,15 @@ class _WebsiteViewState extends State<WebsiteView> {
                       mainAxisAlignment: MainAxisAlignment.center,
                       children: <Widget>[
                         Expanded(
-                          child: FutureBuilder<ImageProvider<dynamic>>(
+                          child: FutureBuilder(
```

Comment:
You can extract this entire FutureBuilder as a widget (something like `WebsiteIcon`) that takes a website as a parameter, and use it in all 3 places to avoid the duplicate code.